### PR TITLE
2169 Support more collate and list of data

### LINKED ISSUE #2169: TransformInverter has `collate` arg but can't work
**Describe the bug**
Currently, the `TransformInverter` hander can't support collate that doesn't generate 1 list of data.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #2169 .

### Description
This PR added support for other `collate` functions in the invert logic and save a list of tensor data.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/transforms/post/dictionary.py [hunk 1]
```diff
@@ -419,6 +419,10 @@ class Invertd(MapTransform):
     A typical usage is to invert the pre-transforms (appplied on input `image`) on the model `pred` data.
 
     Note:
+        According to the `collate_fn`, this transform may return a list of Tensor without batch dim,
+        some following post transforms may not support a list of Tensor, users can leverage the `post_func`
+        arg for basic processing logic.
+
```
Comment (yiheng-wang-nv) [inline monai/transforms/post/dictionary.py:425]: thus some following .... and users can leverage ....

**Review by yiheng-wang-nv — APPROVED** [review]

File: monai/transforms/post/dictionary.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/transforms/post/dictionary.py:425]: updated in the latest commit.
Thanks.

File: monai/transforms/post/dictionary.py [hunk 2]
```diff
@@ -455,8 +459,8 @@ def __init__(
                 default is `meta_dict`, the meta data is a dictionary object.
                 For example, to handle orig_key `image`,  read/write `affine` matrices from the
                 metadata `image_meta_dict` dictionary's `affine` field.
-            collate_fn: how to collate data after inverse transformations.
-                default won't do any collation, so the output will be a list of size batch size.
+            collate_fn: how to collate data after inverse transformations. default won't do any collation,
+                so the output will be a list of PyTorch Tensor or numpy array without batch dim.
```
Comment (rijobro) [inline monai/transforms/post/dictionary.py:463]:
```suggestion
                so the output will be a list of length `batch`, each element consisting of PyTorch Tensor or numpy array without batch dim.
```

**Review by rijobro — APPROVED** [review]

**Pull request merged**